probe-rs/src/probe/jlink/swd.rs

```diff
@@ -633,27 +810,59 @@ impl<Probe: RawSwdIo + 'static> DAPAccess for Probe {
+        let mut succesful_transfers = 0;
 
-        perform_transfers(self, &mut transfers)?;
+        let mut idle_cycles = std::cmp::max(1, self.swd_settings().num_idle_cycles_between_writes);
 
-        for (index, result) in transfers.iter().enumerate() {
-            match &result.status {
-                TransferStatus::Ok => {
-                    values[index] = result.value;
-                }
-                TransferStatus::Failed(err) => {
-                    log::debug!(
-                        "Error in access {}/{} of block access: {}",
-                        index + 1,
-                        values.len(),
-                        err
-                    );
-                    return Err(err.clone().into());
-                }
-                TransferStatus::Pending => {
-                    // This should not happen...
-                    panic!("Error performing transfers")
+        'transfer: for _ in 0..self.swd_settings().num_retries_after_wait {
+            let mut transfers =
+                vec![SwdTransfer::read(port, address); values.len() - succesful_transfers];
+
+            perform_transfers(self, &mut transfers, idle_cycles)?;
+
+            let index_offset = succesful_transfers;
+
+            for (index, result) in transfers.iter().enumerate() {
+                match &result.status {
+                    TransferStatus::Ok => {
+                        values[index_offset + index] = result.value;
+                        succesful_transfers += 1;
+                    }
+                    TransferStatus::Failed(err) => {
+                        log::debug!(
+                            "Error in access {}/{} of block access: {}",
+                            index + 1,
+                            values.len(),
+                            err
+                        );
+
+                        if err == &DapError::WaitResponse {
+                            // Clear STICKORRUN flag
+
+                            // Because we use overrun detection, we now have to clear the overrun error
+                            let mut abort = Abort(0);
+
+                            abort.set_orunerrclr(true);
+
+                            DAPAccess::write_register(
+                                self,
+                                PortType::DebugPort,
+                                Abort::ADDRESS as u16,
+                                abort.into(),
+                            )?;
+
+                            idle_cycles *= 2;
+
+                            log::debug!("Retrying access {}", index_offset + index + 1);
+
+                            continue 'transfer;
+                        }
+                        return Err(err.clone().into());
+                    }
+                    TransferStatus::Pending => {
+                        // This should not happen...
+                        panic!("Error performing transfers")
```

Comment:
I highly dislike this. Is this an actual bug? Could we indicate this in the error message like this:
```suggestion
                        panic!("Error performing transfers. This is a bug and should be reported.")
```